opm/io/eclipse/EGrid.hpp

```diff
@@ -53,27 +54,35 @@ class EGrid : public EclFile
 
     void load_grid_data();
     void load_nnc_data();
+    void mapaxes_init();
+    void mapaxes_transform(double& x, double& y) const;
+    bool with_mapaxes() const { return m_mapaxes_loaded; }
     bool is_radial() const { return m_radial; }
 
     const std::vector<int>& hostCellsGlobalIndex() const { return host_cells; }
     std::vector<std::array<int, 3>> hostCellsIJK();
 
-    // zero based: i1, j1,k1, i2,j2,k2, transmisibility
+    // zero based: i1,j1,k1, i2,j2,k2, transmisibility
     using NNCentry = std::tuple<int, int, int, int, int, int, float>;
     std::vector<NNCentry> get_nnc_ijk();
 
     const std::vector<std::string>& list_of_lgrs() const { return lgr_names; }
 
-    const std::vector<float>& get_mapaxes() const { return m_mapaxes; }
+    const std::array<double, 6>& get_mapaxes() const { return m_mapaxes; }
     const std::string& get_mapunits() const { return m_mapunits; }
 
 private:
     std::filesystem::path inputFileName, initFileName;
     std::string m_grid_name;
     bool m_radial;
 
-    std::vector<float> m_mapaxes;
+    std::array<double, 6> m_mapaxes;
     std::string m_mapunits;
+    mutable bool m_mapaxes_loaded;
```

Comment:
I don't think this needs to be tagged as `mutable`.  As far as I can tell, it is only assigned within the body of the constructor.